src/applications/mhv/secure-messaging/components/NavigationLinks.jsx

```diff
@@ -1,22 +1,66 @@
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { getAllMessages } from '../actions';
 
-const NavigationLinks = () => (
-  <div className="vads-u-text-align--right nav-links">
-    <a
-      className="nav-links-text"
-      href="/message"
-      aria-label="Navigate to previous message"
-    >
-      <i className="fas fa-angle-left" /> Previous
-    </a>
-    <a
-      className="nav-links-text"
-      href="/message"
-      aria-label="Navigate to next message"
-    >
-      Next <i className="fas fa-angle-right" />
-    </a>
-  </div>
-);
+const NavigationLinks = props => {
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { messages } = useSelector(state => state?.allMessages);
+  const messagesIdArr = useRef({});
+  const index = useRef();
+
+  useEffect(() => dispatch(getAllMessages()), [dispatch]);
+  useEffect(
+    () => {
+      if (messages) {
+        messagesIdArr.current = messages.data.map(mes => {
+          return mes.attributes.message_id;
+        });
+
+        index.current = messagesIdArr.current.findIndex(item => {
+          return item === parseInt(props.id, 10);
+        });
+      }
+    },
+    [messages, index, props.id],

```

Comment:
You're updating index inside the useEffect, and also have it in the dependency array, so this useEffect will run at least twice if index gets update.

Reply:
good catch i went ahead and took it off 👍